### PR TITLE
Remove unused allowed unused attributes

### PR DESCRIPTION
**Pull request by lzutao**
changelog: none

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
Have you checked that these don't cause warnings when bootstrapping from the rustc repo?

**Comment by lzutao** [comment]
I haven't. Let's me built with x.py real quick.

**Comment by lzutao** [comment]
Build locally successfully with command: `./x.py test --stage=1 src/tools/clippy`.

File: clippy_lints/src/lib.rs [hunk 1]
```diff
@@ -86,11 +64,6 @@ use rustc_session::Session;
 ///
 /// ```
 /// # #![feature(rustc_private)]
-/// # #[allow(unused_extern_crates)]
-/// # extern crate rustc_middle;
-/// # #[allow(unused_extern_crates)]
-/// # extern crate rustc_session;
-/// # #[macro_use]
```
Comment (lzutao) [inline clippy_lints/src/lib.rs:93]: I pushed a new commit to remove those. Can you re-review?

**Comment by lzutao** [comment]
Oops. I should had have to check the doc tests. Hope it be Okay now!

**Pull request merged**